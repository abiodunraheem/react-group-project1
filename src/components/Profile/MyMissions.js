import { React } from 'react';
import { useSelector } from 'react-redux';
import style from './Profile.module.css';

const MyMissions = () => {
  const missionList = useSelector((state) => state.missionReducer).filter(
    (rocket) => rocket.active === true,
  );
  return (
    <div className={style.mymission}>
      <h1>My Mission</h1>
      <ul className={style.table}>
        {missionList.length > 0 && missionList.map((el) => (
          <li key={el.mission_id}><p>{el.mission_name}</p></li>
        ))}

        {missionList.length < 1 && <li><p>No reserved mission yet</p></li>}
      </ul>
    </div>
  );
};
export default MyMissions;
