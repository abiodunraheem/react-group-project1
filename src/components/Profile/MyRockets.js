import { React } from 'react';
import { useSelector } from 'react-redux';
import style from './Profile.module.css';

const MyRockets = () => {
  const rockets = useSelector((state) => state.RocketsReducer).filter(
    (rocket) => rocket.reserved === true,
  );
  return (
    <div className={style.mymission}>
      <h1>My Rockets</h1>
      <ul className={style.table}>
        {rockets.length > 0 && rockets.map((el) => (
          <li key={el.id}><p>{el.name}</p></li>
        ))}

        {rockets.length < 1 && <li><p>No reserved rocket yet</p></li>}
      </ul>
    </div>
  );
};
export default MyRockets;
