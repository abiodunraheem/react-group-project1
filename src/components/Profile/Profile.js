import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Spinner } from 'react-spinners-css';
import style from './Profile.module.css';
import { displayMissionFunction } from '../../redux/mission/Mission';

const propTypes = {};

const defaultProps = {};

const Profile = () => {
  const dispatch = useDispatch();
  const missionList = useSelector((state) => state.missionReducer);
  useEffect(() => {
    dispatch(displayMissionFunction());
  }, []);
  return (

    <div className={style.myprofile}>
      <ul className={style.myrocket}>
        <li><p>grinade</p></li>
        <li><p>grinade</p></li>
        <li><p>grinade</p></li>
      </ul>
      <ul className={style.myrocket}>
        <li><p>grinade</p></li>
        <li><p>grinade</p></li>
        <li><p>grinade</p></li>
      </ul>
    </div>

  );
};

Profile.propTypes = propTypes;
Profile.defaultProps = defaultProps;

export default Profile;
