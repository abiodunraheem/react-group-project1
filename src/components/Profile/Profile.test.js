import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/ConfigureStore';
import Profile from './Profile';

it('Profile should render', () => {
  const tree = renderer.create(<Provider store={store}><Profile /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
