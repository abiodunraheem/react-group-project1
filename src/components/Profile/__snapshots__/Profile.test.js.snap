// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="myprofile"
>
  <div
    className="mymission"
  >
    <h1>
      My Mission
    </h1>
    <ul
      className="table"
    />
  </div>
  <div
    className="myrocket"
  >
    <h1>
      My Mission
    </h1>
    <ul
      className="table"
    />
  </div>
</div>
`;

exports[`Button component render 1`] = `
<div
  className="myprofile"
>
  <div
    className="mymission"
  >
    <h1>
      My Mission
    </h1>
    <ul
      className="table"
    />
  </div>
  <div
    className="myrocket"
  >
    <h1>
      My Mission
    </h1>
    <ul
      className="table"
    />
  </div>
</div>
`;

exports[`Profile should  1`] = `
<div
  className="myprofile"
>
  <div
    className="mymission"
  >
    <h1>
      My Mission
    </h1>
    <ul
      className="table"
    />
  </div>
  <div
    className="myrocket"
  >
    <h1>
      My Mission
    </h1>
    <ul
      className="table"
    />
  </div>
</div>
`;

exports[`Profile should render 1`] = `
<div
  className="myprofile"
>
  <div
    className="mymission"
  >
    <h1>
      My Mission
    </h1>
    <ul
      className="table"
    />
  </div>
  <div
    className="myrocket"
  >
    <h1>
      My Mission
    </h1>
    <ul
      className="table"
    />
  </div>
</div>
`;
